google_analytics/predictor.py

```diff
@@ -0,0 +1,91 @@
+from abc import abstractmethod
+import numpy as np
+from sklearn.metrics import mean_squared_error
+from sklearn.model_selection import train_test_split, cross_val_score
+from sklearn.base import BaseEstimator, ClassifierMixin
+from utils import timing # noqa
+
+TUNING_OUTPUT_DEFAULT = 'tuning.txt'
+RANDOM_STATE = 42  # Used for reproducible results
+
+
+class Predictor(BaseEstimator, ClassifierMixin):
+    """
+    An abstract class modeling our notion of a predictor.
+    Concrete implementations should follow the predictors
+    interface
+    """
+    name = 'Abstract Predictor'
+
+    def __init__(self, name=name):
+        """
+        Base constructor. The input training is expected to be preprocessed and contain
+        features extracted for each sample along with the true values
+
+        :param params: a dictionary of named model parameters
+        :param name: Optional model name, used for logging
+        """
+        self.name = name
+
+    def __str__(self):
+        return self.name
+
+    @abstractmethod
```

Comment:
Since this guy now has a function body, it shouldn't be abstract anymore (just delete the decoration)